packages/interface/src/components/JoinButton.tsx

```diff
@@ -233,6 +234,20 @@ export const JoinButton = (): JSX.Element => {
         try {
           const parsedPCD = (JSON.parse(result.pcdStr) as jsonPCD).pcd;
           const pcd = await ZKEdDSAEventTicketPCDPackage.deserialize(parsedPCD);
+
+          const response = await fetch("/api/poap", {
+            method: "POST",
+            body: JSON.stringify({ pcd }),
+            headers: {
+              "Content-Type": "application/json",
+            },
+          });
+
+          if (!response.ok) {
+            toast.error("Could not verify your Zupass ticket, please try again");
+          }
+          const { poap } = (await response.json()) as { poap: string };
+          setPoapLink(poap);
```

Comment:
`setPoapLink` is only triggered inside the `handleZupassVerify`, however the `handleZupassVerify` is only triggered when user's not registered and they click the button `Prove you voted with Zupass`. 
So, the outcome is that, if user already signed up, they can never see the `Claim Poap` button.

I think the better way to do this is to 
1. if the user hasn't logged in yet, they click on the login button, generate zupass proof, and if frontend got the pcd and detected the user having the ticket id stored in the contract, display "Claim Poap" button on the main page.
2. if the user already logged in but the frontend doesn't know the pcd, under `view projects`, add a button called `prove you voted and claim your poap`, after user clicking it, it genrates a zupass proof, check if the ticket id stored in the contract, if yes, check the sqlite db, give the user a poap link.

Reply:
what if we remove the login altogether as we don't need it anymore? so rather than performing any checks, we just add the button: "Check if you can claim the voter POAP" and that's when we check the zupass ticket and whether it was signed up to the contract

Reply:
that works for me :)